jwcrypto/jwe.py

```diff
@@ -240,6 +240,55 @@ def unwrap(self, key, keylen, ek, headers):
         return cek
 
 
+class _AesGcmKw(_RawKeyMgmt):
+
+    def __init__(self, keysize):
+        self.backend = default_backend()
+        self.keysize = keysize // 8
+
+    def _get_key(self, key, op):
+        if key.key_type != 'oct':
+            raise InvalidJWEKeyType('oct', key.key_type)
+        rk = base64url_decode(key.get_op_key(op))
+        if len(rk) != self.keysize:
+            raise InvalidJWEKeyLength(self.keysize * 8, len(rk) * 8)
+        return rk
+
+    def wrap(self, key, keylen, cek, headers):
+        rk = self._get_key(key, 'encrypt')
+        if not cek:
+            cek = os.urandom(keylen)
+
+        iv = os.urandom(96 // 8)
+        cipher = Cipher(algorithms.AES(rk), modes.GCM(iv),
+                        backend=self.backend)
+        encryptor = cipher.encryptor()
+        ek = encryptor.update(cek) + encryptor.finalize()
+
+        tag = encryptor.tag
+        return {'cek': cek, 'ek': ek,
+                'header': {'iv': base64url_encode(iv),
+                           'tag': base64url_encode(tag)}}
+
+    def unwrap(self, key, keylen, ek, headers):
+        rk = self._get_key(key, 'decrypt')
+
+        if 'iv' not in headers:
+            raise InvalidJWEData('Invalid Header')
```

Comment:
This exception (and the one below) seem awfully vague. Perhaps you could give more detailed descriptions?